packages/core-app-api/src/apis/implementations/StorageApi/UserSettingsStorage.ts

```diff
@@ -0,0 +1,231 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  DiscoveryApi,
+  ErrorApi,
+  FetchApi,
+  IdentityApi,
+  StorageApi,
+  StorageValueSnapshot,
+} from '@backstage/core-plugin-api';
+import { ResponseError } from '@backstage/errors';
+import { JsonValue, Observable } from '@backstage/types';
+import ObservableImpl from 'zen-observable';
+import { WebStorage } from './WebStorage';
+
+const JSON_HEADERS = {
+  'Content-Type': 'application/json; charset=utf-8',
+  Accept: 'application/json',
+};
+
+const buckets = new Map<string, UserSettingsStorage>();
+
+/**
+ * An implementation of the storage API, that uses the user-settings backend to
+ * persist the data in the DB.
+ *
+ * @public
+ */
+export class UserSettingsStorage implements StorageApi {
```

Comment:
I think this implementation should live in the user settings frontend plugin rather than in the core?

Reply:
Oh quite right! Moved.

(I wonder though, if behavior shouldn't be installed by default in a later release)

Reply:
Perhaps, that'd be up to either create-app or app-defaults though